src/com/xilinx/rapidwright/rwroute/RouteNodeGraph.java

```diff
@@ -170,34 +171,43 @@ public RouteNodeGraph(Design design, RWRouteConfig config) {
         boolean isUltraScale = series == Series.UltraScale;
         boolean isUltraScalePlus = series == Series.UltraScalePlus;
         isVersal = series == Series.Versal;
-        Tile intTile;
         final Set<IntentCode> intTileIntentCodeCareSet;
         Pattern eastWestPattern;
         eastWestWires = new EnumMap<>(TileTypeEnum.class);
         BitSet localWires = new BitSet();
+        List<Tile> intTilesToExamine = null;
         if (isUltraScale || isUltraScalePlus) {
-            intTile = device.getArbitraryTileOfType(TileTypeEnum.INT);
+            Tile tile = device.getArbitraryTileOfType(TileTypeEnum.INT);
             // Device.getArbitraryTileOfType() typically gives you the North-Western-most
             // tile (with minimum X, maximum Y). Analyze the tile just below that.
-            intTile = intTile.getTileXYNeighbor(0, -1);
+            intTilesToExamine = Collections.singletonList(tile.getTileXYNeighbor(0, -1));
             intTileIntentCodeCareSet = EnumSet.of(
                     IntentCode.NODE_PINFEED,
                     IntentCode.NODE_PINBOUNCE,
                     IntentCode.NODE_LOCAL);
 
             ultraScalesLocalWires = new EnumMap<>(TileTypeEnum.class);
-            ultraScalesLocalWires.put(intTile.getTileTypeEnum(), localWires);
+            ultraScalesLocalWires.put(tile.getTileTypeEnum(), localWires);
 
             eastWestPattern = Pattern.compile("(((BOUNCE|BYPASS|IMUX|INODE(_[12])?)_(?<eastwest>[EW]))|INT_NODE_IMUX_(?<inode>\\d+)_).*");
         } else {
             assert(isVersal);
 
             // Find an INT tile adjacent to a CLE_BC_CORE tile since Versal devices may contain AIEs on their northern edge
             Tile bcCoreTile = device.getArbitraryTileOfType(TileTypeEnum.CLE_BC_CORE);
-            // Device.getArbitraryTileOfType() typically gives you the North-Western-most
-            // tile (with minimum X, maximum Y). Analyze the tile just below that.
-            intTile = bcCoreTile.getTileNeighbor(2, 0);
-            assert(intTile.getTileTypeEnum() == TileTypeEnum.INT);
+            Tile tile = bcCoreTile.getTileNeighbor(2, 0);
+            assert(tile.getTileTypeEnum() == TileTypeEnum.INT);
+
+            intTilesToExamine = new ArrayList<>(2);
+            intTilesToExamine.add(tile);
+
+            Tile sllTile = device.getArbitraryTileOfType(TileTypeEnum.SLL);
+            tile = sllTile.getTileNeighbor(2, 0);
```

Comment:
Do we know if all Versal devices have an `SLL` tile type?

Reply:
It should continue to work as before if `SLL` is not present, since there is a `null` check below. 

With that bit of code you just created: can you check which devices have `SLL2` too?

Reply:
Oh wait. I thought I'd put it inside the `null` check. Let me fix that!

Reply:
There is only one device that had an instance of the `SSL2` type:
```
xcvp1902 SLL2_S1X12Y4
```